### PR TITLE
Improving step misalignment after huge step numbers

### PR DESCRIPTION
**Pull request by medghaim**
Using a more precise method to calculate _new_ coordinates for usage in the hex generator. 

## Description
New function to handle the calculation of a new pair of coordinates based on an original location, a distance, and a bearing - `get_new_coords(from_loc, distance, bearing)`

## Motivation and Context
This seriously reduces the misalignment in huge step numbers.

## How Has This Been Tested?
With `-st 120` and `150`. In all quadrants of the earth. This doesn't 100% solve things, but seriously reduces the issue (as seen at the edge of this `-st 120` hex):
image

As opposed to the edge of this `-st 100` hex:
image

Note, the warping only happens in 1 direction, due to map projection. 

The remaining tiny gaps are (_i believe_) floating point imprecision, which can be cleaned up.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by reyqn** [comment]
Yeah this looks way better code wise. I'm closing mine.

**Comment by AHAAAAAAA** [comment]
+1 tested, gaps improved as reported.

**Comment by reyqn** [comment]
@medghaim Actually here is mine at -st 100. (I can't really try more because it's taking very long on my laptop) Does it seem better than yours when you are only at 100?
screenfetch-2016-07-28_00-52-19

**Comment by mwillems1994** [comment]
Perhaps you could change the long if statements for a switch, which is generally faster in comparing integers and makes the code a bit more maintainable.

**Comment by reyqn** [comment]
@AHAAAAAAA Why did you merge this? Here are the results of #2375 for 120 steps.

It's near the north pole and at the most inaccurate place (big latitude and longitude difference from origin)

screenfetch-2016-07-28_01-59-50

We can start seeing little holes but nowhere as important as in this PR.

File: pogom/search.py [hunk 1]
```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
         log.warning("Uncaught exception when downloading map " + str(e))
         return False
 
+def get_new_coords(init_loc, distance, bearing):
+    """ Given an initial lat/lng, a distance(in kms), and a bearing (degrees),
+    this will calculate the resulting lat/lng coordinates.
+    """ 
+    R = 6378.1 #km radius of the earth
```
Comment (reyqn) [inline pogom/search.py]: Try 6378.137 maybe?

**Pull request merged**